csrc/scheduler/matmul.cpp

```diff
@@ -1020,6 +1022,10 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   scheduler_utils::transformPropagateToAllFrom(mma_result, -1);
 
   if (params.use_smem_epilogue) {
+    // TODO: this is no longer needed because we are now using the new swizzle
+    // for the smem epilogue. However, without this, I am seeing some failures
+    // in IdModel. Leaving this as is for now. Will investigate later.
+
```

Comment:
I guess I don't have to do `swizzleSharedMemory`. As a workaround, I guess I just need to use some bogus transformations (maybe split 1 then merge back?) to prevent mapping of compliment IDs in the permissive graph causing loop domains mapped with each other. I don't want to work on this in this PR. I might spent a small amount of time cleaning this up in a future PR. But I do believe we should start the process of removing forwarding. This process can be low priority, but I'd like to see it slowly making progress.

Reply:
Example failure:

```C++
[ RUN      ] GPUTTensorCoreTest.FusionAmpereMatmulSplitKBias_CUDA
unknown file: Failure
C++ exception with description "!self_mappped_loop_pair.has_value() INTERNAL ASSERT FAILED at "/home/gaoxiang/Fuser/csrc/id_model/id_model.cpp":846, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Detected loop domains are mapped in the loop graph. Tensor: T3_l[ bS64{( ceilDiv(1, 128) )}, iS66{( ceilDiv(( (( (( getMetaData(T1) )).logical_size ))[2] ), 128) )}, iS70{2}, iS71{( ceilDiv(( ceilDiv(( (( (( getMetaData(T0) )).logical_size ))[2] ), 32) ), 2) )}, iS142{( ceilDiv(32, 16) )}, bthreadIdx.z144{( ceilDiv(128, 64) )}, ithreadIdx.y146{( ceilDiv(128, 64) )}, bS148{( ceilDiv(64, 16) )}, iS150{( ceilDiv(64, 8) )}, bS149{16}, ithreadIdx.x228{( 2 * ( ( ( ceilDiv(( ceilDiv(16, 2) ), 4) ) * ( ceilDiv(8, 8) ) ) * ( 4 * 2 ) ) )}, iV227{( ceilDiv(8, 2) )} ] ca_pos( 7 ) produce_pos( 4 ). Mapped loop domains: bthreadIdx.z144{( ceilDiv(128, 64) )} and ithreadIdx.y146{( ceilDiv(128, 64) )}
Exception raised from validateLoopGraphHasNoSelfMappedLeafDomains at /home/gaoxiang/Fuser/csrc/id_model/id_model.cpp:846 (most recent call first):
frame #0: <unknown function> + 0xa311c (0x654c848a211c in ./bin/test_matmul)
frame #1: <unknown function> + 0x2f1a8c (0x654c84af0a8c in ./bin/test_matmul)
frame #2: <unknown function> + 0x3dffff (0x654c84bdefff in ./bin/test_matmul)
frame #3: <unknown function> + 0x3e9447 (0x654c84be8447 in ./bin/test_matmul)
frame #4: <unknown function> + 0x3e99d6 (0x654c84be89d6 in ./bin/test_matmul)
frame #5: <unknown function> + 0x3eaaf8 (0x654c84be9af8 in ./bin/test_matmul)
frame #6: <unknown function> + 0x202d25 (0x654c84a01d25 in ./bin/test_matmul)
frame #7: <unknown function> + 0x204bce (0x654c84a03bce in ./bin/test_matmul)
frame #8: <unknown function> + 0x307c3c (0x654c84b06c3c in ./bin/test_matmul)
frame #9: <unknown function> + 0x7b23c9 (0x654c84fb13c9 in ./bin/test_matmul)
frame #10: <unknown function> + 0x7a429c (0x654c84fa329c in ./bin/test_matmul)
frame #11: <unknown function> + 0x8656f7 (0x654c850646f7 in ./bin/test_matmul)
frame #12: <unknown function> + 0x84e516 (0x654c8504d516 in ./bin/test_matmul)
frame #13: <unknown function> + 0x84e755 (0x654c8504d755 in ./bin/test_matmul)
frame #14: <unknown function> + 0x84effb (0x654c8504dffb in ./bin/test_matmul)
frame #15: <unknown function> + 0x85cb65 (0x654c8505bb65 in ./bin/test_matmul)
frame #16: <unknown function> + 0x865d87 (0x654c85064d87 in ./bin/test_matmul)
frame #17: <unknown function> + 0x84e83f (0x654c8504d83f in ./bin/test_matmul)
frame #18: <unknown function> + 0x13b1ab (0x654c8493a1ab in ./bin/test_matmul)
frame #19: <unknown function> + 0x25c88 (0x78dff0e39c88 in /usr/lib/libc.so.6)
frame #20: __libc_start_main + 0x8c (0x78dff0e39d4c in /usr/lib/libc.so.6)
frame #21: <unknown function> + 0x13b585 (0x654c8493a585 in ./bin/test_matmul)
" thrown in the test body.

To reproduce: NVFUSER_TEST_RANDOM_SEED=1721778791 NVFUSER_TEST_ATEN_RANDOM_SEED=0 nvfuser_tests --gtest_filter='GPUTTensorCoreTest.FusionAmpereMatmulSplitKBias_CUDA'
[  FAILED  ] GPUTTensorCoreTest.FusionAmpereMatmulSplitKBias_CUDA (943 ms)
[ RUN      ] GPUTTensorCoreTest.FusionAmpereMatmulBatchSplitK_CUDA
terminate called after throwing an instance of 'nvfuser::nvfError'
  what():  _result == CUDA_SUCCESS INTERNAL ASSERT FAILED at "/home/gaoxiang/Fuser/csrc/executor_utils.cpp":1152, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. CUDA error: CUDA_ERROR_ILLEGAL_ADDRESS failed with error an illegal memory access was encountered
Exception raised from ~CompiledKernel at /home/gaoxiang/Fuser/csrc/executor_utils.cpp:1152 (most recent call first):
frame #0: <unknown function> + 0xa311c (0x654c848a211c in ./bin/test_matmul)
frame #1: <unknown function> + 0x2f1a8c (0x654c84af0a8c in ./bin/test_matmul)
frame #2: <unknown function> + 0x329806 (0x654c84b28806 in ./bin/test_matmul)
frame #3: <unknown function> + 0x3d648e (0x654c84bd548e in ./bin/test_matmul)
frame #4: <unknown function> + 0x11b372 (0x654c8491a372 in ./bin/test_matmul)
frame #5: <unknown function> + 0x8656f7 (0x654c850646f7 in ./bin/test_matmul)
frame #6: <unknown function> + 0x84e516 (0x654c8504d516 in ./bin/test_matmul)
frame #7: <unknown function> + 0x84e755 (0x654c8504d755 in ./bin/test_matmul)
frame #8: <unknown function> + 0x84effb (0x654c8504dffb in ./bin/test_matmul)
frame #9: <unknown function> + 0x85cb65 (0x654c8505bb65 in ./bin/test_matmul)
frame #10: <unknown function> + 0x865d87 (0x654c85064d87 in ./bin/test_matmul)
frame #11: <unknown function> + 0x84e83f (0x654c8504d83f in ./bin/test_matmul)
frame #12: <unknown function> + 0x13b1ab (0x654c8493a1ab in ./bin/test_matmul)
frame #13: <unknown function> + 0x25c88 (0x78dff0e39c88 in /usr/lib/libc.so.6)
frame #14: __libc_start_main + 0x8c (0x78dff0e39d4c in /usr/lib/libc.so.6)
frame #15: <unknown function> + 0x13b585 (0x654c8493a585 in ./bin/test_matmul)


Aborted (core dumped)
```